lib/matplotlib/_pylab_helpers.py

```diff
@@ -97,6 +97,19 @@ def get_active(cls):
         """Return the active manager, or *None* if there is no manager."""
         return next(reversed(cls.figs.values())) if cls.figs else None
 
+    @classmethod
+    def _set_new_active_manager(cls, manager):
+        """Adopt *manager* into pyplot and make it the active manager."""
+        if not hasattr(manager, "_cidgcf"):
+            manager._cidgcf = manager.canvas.mpl_connect(
+                "button_press_event", lambda event: cls.set_active(manager))
```

Comment:
(Alternatively, this callback could be made "pickling-persistent" using the mechanism proposed in https://github.com/matplotlib/matplotlib/pull/16220.)